python_modules/dagster-graphql/dagster_graphql/schema/backfill.py

```diff
@@ -532,6 +538,67 @@ def resolve_title(self, _graphene_info: ResolveInfo) -> Optional[str]:
     def resolve_description(self, _graphene_info: ResolveInfo) -> Optional[str]:
         return self._backfill_job.description
 
+    def resolve_logEvents(self, graphene_info: ResolveInfo, cursor: Optional[str] = None):
+        from ..schema.instigation import (
+            GrapheneInstigationEvent,
+            GrapheneInstigationEventConnection,
+        )
+        from ..schema.logs.log_level import GrapheneLogLevel
+
+        backfill_log_key_prefix = self._backfill_job.log_storage_prefix
+
+        instance = graphene_info.context.instance
+
+        if not isinstance(instance.compute_log_manager, CapturedLogManager):
+            return GrapheneInstigationEventConnection(events=[], cursor="", hasMore=False)
+
+        if not instance.backfill_log_storage_enabled():
+            return GrapheneInstigationEventConnection(events=[], cursor="", hasMore=False)
+
+        records, new_cursor = instance.compute_log_manager.read_log_lines_for_log_key_prefix(
+            backfill_log_key_prefix, cursor
+        )
+
+        events = []
+        for line in records:
+            if not line:
+                continue
+            try:
+                record_dict = _seven.json.loads(line)
+            except json.JSONDecodeError:
+                continue
+
+            exc_info = record_dict.get("exc_info")
+            message = record_dict.get("msg")
+            if exc_info:
+                message = f"{message}\n\n{exc_info}"
+            event = GrapheneInstigationEvent(
+                message=message,
+                level=GrapheneLogLevel.from_level(record_dict["levelno"]),
+                timestamp=int(record_dict["created"] * 1000),
+            )
+
+            events.append(event)
+
+        if (
+            new_cursor
+            and not new_cursor.has_more
+            and self.status
+            in [
+                GrapheneBulkActionStatus.COMPLETED,
+                GrapheneBulkActionStatus.FAILED,
+                GrapheneBulkActionStatus.CANCELED,
+            ]
+        ):
+            has_more = False
+        else:
+            has_more = True
```

Comment:
I think we should remove the self.status check from this logic -- right now, if the action is in progress, it looks like we'll return has_more=True continuously.

I think this is a good idea (we do want it to make another request and ask for more logs), but the UI will probably want to fetch the next page "immediately" if the cursor is behind, and "in 5 seconds" if it's up to the latest cursor and the backfill is in progress.  Because of the different delays, it might be better to put this part of the logic on the front-end.

Reply:
ok sounds good to me - made the update